pkg/codegen/nodejs/gen.go

```diff
@@ -670,20 +670,24 @@ func (mod *modContext) genFunction(w io.Writer, fun *schema.Function) {
 	}
 }
 
-func visitObjectTypes(t schema.Type, visitor func(*schema.ObjectType)) {
+func visitObjectTypes(t schema.Type, visitor func(*schema.ObjectType), seen map[schema.Type]struct{}) {
```

Comment:
I think `codegen.Set` works here.

```suggestion
func visitObjectTypes(t schema.Type, visitor func(*schema.ObjectType), seen codegen.Set) {
```